test/src/components/year-selector2.test.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import {render} from '@testing-library/preact';
+
+// The date needs to be set before the import
+jest.useFakeTimers().setSystemTime(new Date('2024-08-04'));
+describe('year-selector', () => {
+    it('starts with 1 year back when after July', async () => {
```

Comment:
doesn't really matter but I think "after June" or "during or after July" would be more accurate for someone trying to understand the rules in the future

Reply:
Fair. I went with before July/after June